acp/acp.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+/*
+Package acp utilizes the sourcehub acp module to bring the functionality to defradb, this package also helps
+avoid the leakage of direct sourcehub references through out the code base, and eases in swapping
+between local embedded use case and a more global on sourcehub use case.
+*/
+
+package acp
+
+import (
+	"context"
+
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/logging"
+)
+
+var (
+	log = logging.MustNewLogger("acp")
+
+	NoACPModule = immutable.None[ACPModule]()
+)
+
+// ACPModule is the interface to all types of access control modules there might exist.
+//
+// An embedded local implementation might restrict the implementation to not make remote calls.
+//
+// A remote implementation might allow remote calls to access the acp module in the cloud.
+// Note: sourcehub acp module will likely need the remote implementation.
+type ACPModule interface {
+
+	// Start initializes the acp module, with persistence data stored under the given folder.
+	//
+	// If previously started in the same path, then recovers / reloads (unless used in-memory).
+	Start(context.Context, string) error
```

Comment:
todo: Please document whether this is limited to relative or absolute directory paths.  And maybe also as to whether it will create/truncate and files in the given location.

Reply:
- [ ] document abs/relative path on acp `Start()`